src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseChecklistService.cs

```diff
@@ -143,28 +149,31 @@ private async Task<bool> ReleaseHasNonEmptyHeadlineBlock(Guid releaseId)
 
         private async Task<bool> ReleaseHasEmptyGenericContentSection(Guid releaseId)
         {
-            return await _contentDbContext.ReleaseContentSections
-                .Include(rcs => rcs.ContentSection)
-                .ThenInclude(cs => cs.Content)
-                .Where(rcs =>
-                    rcs.ReleaseId == releaseId
-                    && rcs.ContentSection.Type == ContentSectionType.Generic)
-                .AnyAsync(rcs => rcs.ContentSection.Content.Count == 0);
+            return await _contentDbContext.ContentSections
+                .Where(cs =>
+                    cs.Type == ContentSectionType.Generic && 
+                    cs.Release.ReleaseId == releaseId)
+                .AnyAsync(cs => cs.Content.Count == 0);
         }
 
         private async Task<bool> ReleaseGenericContentSectionsContainEmptyContentBlock(Guid releaseId)
         {
-            var releaseGenericContentBlocks = await _contentDbContext.ReleaseContentSections
-                .Include(rcs => rcs.ContentSection)
-                .ThenInclude(cs => cs.Content)
-                .Where(rcs =>
-                    rcs.ReleaseId == releaseId
-                    && rcs.ContentSection.Type == ContentSectionType.Generic)
-                .SelectMany(rcs => rcs.ContentSection.Content)
-                .ToListAsync();
+            return await _contentDbContext.ContentBlocks
+                .Where(cb =>
+                    cb.ContentSection!.Type == ContentSectionType.Generic && 
+                    cb.ContentSection.Release.ReleaseId == releaseId)
+                .OfType<HtmlBlock>()
+                .AnyAsync(htmlBlock => string.IsNullOrEmpty(htmlBlock.Body));
+        }
 
-            return releaseGenericContentBlocks
-                .Any(block => block is HtmlBlock htmlBlock && htmlBlock.Body.IsNullOrEmpty());
+        private async Task<bool> ReleaseRelatedDashboardsSectionContainsEmptyContentBlock(Guid releaseId)
+        {
+            return await _contentDbContext.ContentBlocks
+                .Where(cb =>
+                    cb.ContentSection!.Type == ContentSectionType.RelatedDashboards && 
+                    cb.ContentSection.Release.ReleaseId == releaseId)
+                .OfType<HtmlBlock>()
+                .AnyAsync(htmlBlock => string.IsNullOrEmpty(htmlBlock.Body));
```

Comment:
Not particularly fussed about this, but you could accept a `ContentSectionType` as an argument and merge these two methods into one.

Reply:
I think maybe `ReleaseHasNonEmptyHeadlineBlock` could also be removed as part of this - possibly not because I think that also ensures all headlines blocks are `HtmlBlock`s.

Reply:
The error produced by `ReleaseHasNonEmptyHeadlineBlock` needs to check there's a html block with content as opposed to checking there's no empty html blocks. I've added a new method `ReleaseSectionHasNonEmptyHtmlBlock` to do that and changed 

```
(!await ReleaseHasKeyStatistic(release.Id) &&
 !await ReleaseHasNonEmptyHeadlineBlock(release.Id))
```

to

```
if (!(await ReleaseHasKeyStatistic(release.Id) ||
      await ReleaseSectionHasNonEmptyHtmlBlock(release.Id, ContentSectionType.Headlines)))
```